packages/multichain-transactions-controller/src/MultichainTransactionsController.ts

```diff
@@ -285,16 +280,35 @@ export class MultichainTransactionsController extends BaseController<
           { limit: 10 },
         );
 
-        const transactions = this.#filterTransactions(response.data);
+        const transactionsByChain: Record<string, Transaction[]> = {};
+
+        response.data.forEach((transaction) => {
+          const chain = transaction.chain as string;
```

Comment:
I think this should already be a `CaipChainId` no? So we shouldn't need to cast as a `string` IMO.

```suggestion
          const { chain } = transaction;
```